src/python/pants/backend/python/targets/unpacked_whls.py

```diff
@@ -57,15 +58,21 @@ def __init__(
         :param compatibility: Python interpreter constraints used to create the pex for the requirement
                               target. If unset, the default interpreter constraints are used. This
                               argument is unnecessary unless the native code depends on libpython.
-        :param str within_data_subdir: If provided, descend into '<name>-<version>.data/<subdir>' when
-                                       matching `include_patterns`. For python wheels which declare any
-                                       non-code data, this is usually needed to extract that without
-                                       manually specifying the relative path, including the package
-                                       version. For example, when `data_files` is used in a setup.py,
-                                       `within_data_subdir='data'` will allow specifying
-                                       `include_patterns` matching exactly what is specified in the
-                                       setup.py.
+        :param bool within_data_subdir: If True, descend into '<name>-<version>.data/' when matching
+                                        `include_patterns`. For python wheels which declare any non-code
+                                        data, this is usually needed to extract that without manually
+                                        specifying the relative path, including the package version. For
+                                        example, when `data_files` is used in a setup.py,
+                                        `within_data_subdir=True` will allow specifying
+                                        `include_patterns` matching exactly what is specified in the
+                                        setup.py.
         """
+        deprecated_conditional(
+            lambda: type(within_data_subdir) not in (bool, type(None)),
+            removal_version="1.28.0.dev2",
```

Comment:
We clarified in https://github.com/pantsbuild/pants/pull/8908 that you are allowed to target `1.28.0.dev0` or `1.28.0.dev1` if you'd like.

It's fine too if you still want to target `1.28.0.dev2.

Reply:
Aha. I hadn't tracked that but will leave as-is. I'm in no rush on any of these.